csharp.html

```diff
@@ -21,21 +22,41 @@
   <% end %>
   <% end %>
   <% end %>
-
-  <% if @method in ['GET', 'DELETE']: %>
-  var response = await httpClient.<%- @method.charAt(0).toUpperCase() + @method.substring(1).toLowerCase() %>Async("<%- @url %>");
-  <% else if @method in ['POST', 'PUT']: %>
-
-  <% if @content_type? : %>
-  var content = new StringContent(<%- @helpers.escape(@body) %>, System.Text.Encoding.Default, "<%- @content_type %>");
-  <% else : %>
-  var content = new StringContent(<%- @helpers.escape(@body) %>);
-  <% end %>
-
-  var response = await httpClient.<%- @method.charAt(0).toUpperCase() + @method.substring(1).toLowerCase() %>Async("<%- @url %>", content);
-  <% end %>
-
-  string responseData = await response.Content.ReadAsStringAsync();
-
-}
+  <% if @method in ['GET', 'DELETE', 'OPTIONS', 'HEAD']: %>
+  <% if @method is 'HEAD': %>
+  var requestUri = new Uri(baseAddress, "<%- @url.substring(1) %>");
```

Comment:
Please always use `<%=` when printing user-input data.

For all `<%- @url.substring(1) %>` in lines below too, please. @XVincentX